compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -710,16 +702,20 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         let source_info = self.source_info(span);
         self.cfg.push(block, Statement { source_info, kind: StatementKind::StorageLive(local_id) });
         let region_scope = self.region_scope_tree.var_scope(var.local_id);
-        if schedule_drop {
-            self.schedule_drop(span, region_scope, local_id, DropKind::Storage);
+        // Altough there is almost always scope for given variable in corner cases
+        // like #92893 we might get variable with no scope.
+        if region_scope.is_some() && schedule_drop {
+            self.schedule_drop(span, region_scope.unwrap(), local_id, DropKind::Storage);
         }
         Place::from(local_id)
     }
 
     crate fn schedule_drop_for_binding(&mut self, var: HirId, span: Span, for_guard: ForGuard) {
         let local_id = self.var_local_id(var, for_guard);
         let region_scope = self.region_scope_tree.var_scope(var.local_id);
-        self.schedule_drop(span, region_scope, local_id, DropKind::Value);
+        if region_scope.is_some() {
```

Comment:
Please use `if let Some(var_scope)` instead of unwrapping

The same in the other usage above